"false" undefined in charset.c

Hi,

I just tried to compile 3.12.0 on an EL9 and found that the recently introduced function `static bool ucharbuf_putc32(struct ucharbuf *ubuf, uint32_t c)` is using `bool`/`false` which the compiler doesn't understand:

```
lib/charset.c:2206:8: error: unknown type name 'bool'
 2206 | static bool ucharbuf_putc32(struct ucharbuf *ubuf, uint32_t c)
      |        ^~~~
lib/charset.c: In function 'ucharbuf_putc32':
lib/charset.c:2210:22: error: 'false' undeclared (first use in this function)
 2210 |     UBool is_error = false;
      |                      ^~~~~
[…]
```
Adding `#include <stdbool.h>` in `lib/charset.c` worked for me.

--- a/lib/charset.c
+++ b/lib/charset.c
@@ -41,6 +41,7 @@
  */
 
 #include <config.h>
+#include <stdbool.h>
 #include <stdlib.h>
 #include <string.h>
 #include <syslog.h>

